lib/modules/manager/pipenv/artifacts.spec.ts

```diff
@@ -328,4 +332,72 @@ describe('modules/manager/pipenv/artifacts', () => {
     ).not.toBeNull();
     expect(execSnapshots).toMatchSnapshot();
   });
+
+  it('passes private credential environment vars', async () => {
+    fs.readLocalFile.mockResolvedValueOnce('current pipfile');
+    const packageFileContent = {
+      registryUrls: ['https://$USERNAME:${PASSWORD}@mypypi.example.com/simple'],
+      deps: [],
+    };
+    pipenv.extractPackageFile.mockResolvedValueOnce(packageFileContent);
+    find.mockReturnValueOnce({
+      username: 'usernameOne',
+      password: 'passwordTwo',
+    });
+
+    fs.ensureCacheDir.mockResolvedValueOnce(
+      '/tmp/renovate/cache/others/pipenv'
+    );
+    fs.readLocalFile.mockResolvedValueOnce('current pipfile.lock');
+    const execSnapshots = mockExecAll();
+    git.getRepoStatus.mockResolvedValue(
+      partial<StatusResult>({
+        modified: ['Pipfile.lock'],
+      })
+    );
+    fs.readLocalFile.mockResolvedValueOnce('New Pipfile.lock');
+
+    expect(
+      await pipenv.updateArtifacts({
+        packageFileName: 'Pipfile',
+        updatedDeps: [],
+        newPackageFileContent: 'some new content',
+        config: { ...config, constraints: { python: '== 3.8.*' } },
+      })
+    ).not.toBeNull();
+    expect(execSnapshots).toMatchSnapshot();
```

Comment:
use toMatchObject

Reply:
Could you point me where it's implemented that way. I just took an inspiration (aka copy-pasted) another test case from the same manager.

Reply:
@viceice @rarkins Could you point me where it's implemented that way. I just took an inspiration (aka copy-pasted) another test case from the same manager. And there are no test cases in `pipenv` manager that use these expectations.

Reply:
I think this is a similar example: https://github.com/renovatebot/renovate/blob/0d758a3d0e24727bd7a383b3ed98a7a7a9abe0f7/lib/modules/manager/npm/post-update/npm.spec.ts#L289-L314